modules/govuk_elasticsearch/manifests/snapshot.pp

```diff
@@ -0,0 +1,71 @@
+# define govuk_elasticsearch::snapshot
+#
+# [*s3_bucket]
+#   The S3 bucket where the snapshots will be stored
+#
+# [*s3_region]
+#   The AWS region for the S3 bucket. Defaults to eu-west-1.
+#
+# [*s3_path]
+#   The folder path within the S3 bucket where the snapshots
+#   will be stored
+#
+# [*index]
+#   The ES index to backup
+#
+# [*http_port]
+#   The port that ES runs on
+#
+# [*hour]
+#   The hour to start the snapshot (crontab)
+#
+# [*minute]
+#   The minute to run the snapshot (crontab)
+#
+# [*title*]
+#   The $title or $name of the hiera hash should be the index name
+#
+# Hiera example:
+# --------------
+# govuk_elasticsearch::snapshot::indices:
+#   kibana-int:
+#     s3_bucket: 'jonauman-test2'
+#     s3_path: 'elasticsearch'
+
+define govuk_elasticsearch::snapshot
+  ($s3_bucket = undef,
+  $s3_region = 'eu-west',
+  $s3_path = '',
```

Comment:
I _think_ an empty string default would cause the `<% if @s3_path %>` statement below to trigger anyway, while an `undef` would not (although I'm not 100%).